ts/src/pro/test/Exchange/test.watchOrderBook.ts

```diff
@@ -1,65 +1,51 @@
-'use strict';
-
-// ----------------------------------------------------------------------------
-
 
+import assert from 'assert';
 import testOrderBook from '../../../test/Exchange/base/test.orderBook.js';
-import errors from '../../../base/errors.js';
-
-/*  ------------------------------------------------------------------------ */
-
-export default async (exchange, symbol) => {
-
-    // log (symbol.green, 'watching order book...')
+import testSharedMethods from '../../../test/Exchange/base/test.sharedMethods.js';
 
+async function testWatchOrderBook (exchange, skippedProperties, symbol) {
     const method = 'watchOrderBook';
-    const skippedProperties = {};
-
-    // we have to skip some exchanges here due to the frequency of trading or to other factors
-    const skippedExchanges = [
-        'cex', // requires authentication
-        'kucoin', // requires authentication for public orderbooks
-        'luno', // requires authentication for public orderbooks
-        'ripio',
-        'gopax', // requires authentication for public orderbooks
-        'woo',
-        'alpaca', // requires auth
-        'coinbasepro', // requires auth
-        'coinbaseprime', // requires auth
-    ];
-
-    if (skippedExchanges.includes (exchange.id)) {
-        console.log (exchange.id, method, '() test skipped');
-        return;
-    }
-
-    if (!exchange.has[method]) {
-        console.log (exchange.id, 'does not support', method, '() method');
-        return;
-    }
-
-    let response = undefined;
-
-    let now = Date.now ();
-    const ends = now + 10000;
-
+    let now = exchange.milliseconds ();
+    const ends = now + 15000;
     while (now < ends) {
-
+        let response = undefined;
         try {
-
-            response = await exchange[method] (symbol);
-
-            testOrderBook (exchange, skippedProperties, method, response, symbol);
-
+            response = await exchange.watchOrderBook (symbol);
         } catch (e) {
-
-            if (!(e instanceof errors.NetworkError)) {
+            if (!testSharedMethods.isTemporaryFailure (e)) {
                 throw e;
             }
+            now = exchange.milliseconds ();
+            continue;
         }
-
-        now = Date.now ();
+        [ response, skippedProperties ] = fixPhpObjectArray (exchange, response, skippedProperties);
+        assert (typeof response === 'object', exchange.id + ' ' + method + ' ' + symbol + ' must return an object. ' + exchange.json (response));
+        now = exchange.milliseconds ();
+        testOrderBook (exchange, skippedProperties, method, response, symbol);
     }
-
-    return response;
-};
+}
+
+function fixPhpObjectArray (exchange, response, skippedProperties) {
```

Comment:
I still don't like this :/, first does not replicate the behavior on the user side, second is not concurrent safe, third is PHP only. 

Can't we comment out the failing assertion in PHP for now, instead of adding this?

Reply:
We remove this entirely, and do a custom regex replace to replace the failing assertion in PHP only,  no?

Reply:
Ok, i'll do it as to skip those orderbook tests in PHP temporarily